Python/Product/Analysis/Values/BoundBuiltinMethodInfo.cs

```diff
@@ -50,7 +51,12 @@ public BoundBuiltinMethodInfo(IPythonBoundFunction function, PythonAnalyzer proj
         }
 
         public override IAnalysisSet Call(Node node, AnalysisUnit unit, IAnalysisSet[] args, NameExpression[] keywordArgNames) {
-            return _method.ReturnTypes.GetInstanceType();
+            // Check if method returns self
+            var returnType = _method.ReturnTypes.GetInstanceType();
+            if (args.Length > 0 && returnType.FirstOrDefault() is BuiltinInstanceInfo biif && biif.PythonType == _method.Function?.DeclaringType) {
```

Comment:
There's a `.Split` extension method for `returnType` here that makes it easier to handle multiple return values properly (split with a predicate, then union and return). Otherwise, I like the approach